### PR TITLE
feat: support resetting TileLayer

### PR DESCRIPTION
**Pull request by tprebs**
This change supports the clearing of in memory tiles from the TileLayerState as described in  #666 

I created a separate Stream to manage reset events instead of using the LayerOptions rebuild stream as this is currently consumed by the onMove event subscriber and any changes to this may affect existing implementations using the rebuild. 

I have also added an example for resetting the a TileLayer

### CODE REVIEW

**Comment by JaffaKetchup** [comment]
Any news?

**Comment by tprebs** [comment]
@JaffaKetchup, I can bring this PR inline with master this weekend if required. I haven't looked at this in a few months

**Comment by JaffaKetchup** [comment]
Don't worry about it, it would just be nice to have, and improve the experience of my plugin especially.

**Comment by tonyshkurenko** [comment]
@tprebs would be great!

**Comment by JaffaKetchup** [comment]
@tprebs If you're busy, I can bring it up to date now for you?

**Comment by tonyshkurenko** [comment]
I would love to have this merged!

**Comment by JaffaKetchup** [comment]
Don't like to ask again, but any hope of this coming soon? @tprebs

**Comment by tprebs** [comment]
@JaffaKetchup sorry for the delay, it's been a busy couple of months. I have brought the PR up to master. I have also been planning on making the feaflet heatmap plugin available but there are just not enough hours in the day

**Review by kengu — APPROVED** [review]
LGTM. Thanks for the contribution!

**Comment by JaffaKetchup** [comment]
@tprebs Thanks, and don't worry about the delay, it's similar story for me. Great that you've finally found time to do it!

**Comment by JaffaKetchup** [comment]
@kengu Thanks for all the merges today :), is there a pub.dev update planned any time soon to publish these changes?

PS. You haven't merged #989, correct me if I'm wrong, but I'm assuming you need another maintainers approval as it was originally written by you?

**Comment by kengu** [comment]
Yes, I'm not allowed to approve my own PRs, so we need to wait on @johnpryan or some of the other contributors with write access to master.

**Pull request merged**
